tools/c7n_kube/c7n_kube/actions.py

```diff
@@ -45,3 +49,80 @@ def process_resource_set(self, client, model, resources):
         for r in resources:
             log.info('%s %s' % (op_name, r))
         pass
+
+
+class PatchAction(MethodAction):
+    """
+    Patches a resource
+
+    Requires patch and namespaced attributes on the resource definition
+    """
+    def validate(self):
+        if not self.manager.get_model().patch:
+            raise PolicyValidationError('patch attribute not defined for resource')
+        return self
+
+    def get_permissions(self):
+        patch = self.manager.get_model().patch
+        return ''.join([a.capitalize() for a in patch.split('_')])
+
+    def patch_resources(self, op, patch_args, namespaced, resources):
```

Comment:
most of the parameters (op, namespaced) here seem redundant, actions have access to all this information from the resource manager

Reply:
agreed on namespaced, but op isn't available in the resource manager as I need to grab the function off the client class

Reply:
right, but just pass the client, it makes the signature more generic, the impl can retrieve the op using the resource type metadata here, also removes ambiguity passing around op, when it could construed as method or string metadata.